namesrv/src/main/java/org/apache/rocketmq/namesrv/kvconfig/KVConfigManager.java

```diff
@@ -62,27 +62,27 @@ public void load() {
     public void putKVConfig(final String namespace, final String key, final String value) {
         try {
             this.lock.writeLock().lockInterruptibly();
-            try {
-                HashMap<String, String> kvTable = this.configTable.get(namespace);
-                if (null == kvTable) {
-                    kvTable = new HashMap<>();
-                    this.configTable.put(namespace, kvTable);
-                    log.info("putKVConfig create new Namespace {}", namespace);
-                }
+            HashMap<String, String> kvTable = this.configTable.get(namespace);
+            if (null == kvTable) {
+                kvTable = new HashMap<>();
+                this.configTable.put(namespace, kvTable);
+                log.info("putKVConfig create new Namespace {}", namespace);
+            }
 
-                final String prev = kvTable.put(key, value);
-                if (null != prev) {
-                    log.info("putKVConfig update config item, Namespace: {} Key: {} Value: {}",
-                        namespace, key, value);
-                } else {
-                    log.info("putKVConfig create new config item, Namespace: {} Key: {} Value: {}",
-                        namespace, key, value);
-                }
-            } finally {
-                this.lock.writeLock().unlock();
+            final String prev = kvTable.put(key, value);
+            if (null != prev) {
+                log.info("putKVConfig update config item, Namespace: {} Key: {} Value: {}",
+                    namespace, key, value);
+            } else {
+                log.info("putKVConfig create new config item, Namespace: {} Key: {} Value: {}",
+                    namespace, key, value);
             }
         } catch (InterruptedException e) {
             log.error("putKVConfig InterruptedException", e);
+            // set interrupt flag
+            Thread.currentThread().interrupt();
+        } finally {
+            this.lock.writeLock().unlock();
```

Comment:
IllegalMonitorStateException will be thrown if lock failed?

Reply:
Thank you for your reply,
my idea was wrong, I did not think that the lock will throw exception after the lock failure

Reply:
anyway, thanks for your contribution~